MyEnumTesting/MyEnumerableTests.cs

```diff
@@ -0,0 +1,259 @@
+﻿using Xunit;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using System;
+using Sol5_1_Collections;
+using System.Linq;
+using System.Diagnostics;
+
+namespace MyEnumTesting
+{
+    public class MyEnumerableTests
+    {
+        [Fact]
+        public async void Run_MyEnumerable_FiftyValidTasks_ResultExpected()
+        {
+            var funcs = Enumerable.Range(1, 50)
+            .Select(i => (Func<CancellationToken, Task<int>>)(async (CancellationToken cs) => await GetResultWithoutExceptions(i, cs)));
+
+            var myEnum = new MyAsyncEnumerable<int>(funcs);
+
+            var (results, exceptions) = await GetResult_From_RunAwaitForeach(myEnum);
+            results.Sort();
+            Assert.Equal(actual: exceptions.InnerExceptions.Count, expected: 0);
+            Assert.Equal(actual: results, expected: Enumerable.Range(1, 50).ToList());
+        }
+
+        [Fact]
+        public async void Run_MyEnumerable_OneException_ResultExpected()
+        {
+            var funcs = Enumerable.Range(1, 50)
+            .Select(i => (Func<CancellationToken, Task<int>>)(async (CancellationToken cs) => await GetResultWithOneException(i, cs)));
+
+            var myEnum = new MyAsyncEnumerable<int>(funcs);
+            var artificialResult = Enumerable.Range(2, 49).ToList();
+            var arificialAggEx = new AggregateException(new Exception("Ошибка при получении поста номер: 1"));
+
+            var (results, exceptions) = await GetResult_From_RunAwaitForeach(myEnum);
+            results.Sort();
+
+            Assert.Equal(actual: results, expected: artificialResult);
+            Assert.True(CompareAggregateEx(exceptions, arificialAggEx));
+        }
+
+        [Fact]
+        public async void Run_MyEnumerable_MultipleExceptions_ResultExpected()
+        {
+            var funcs = Enumerable.Range(1, 50)
+            .Select(i => (Func<CancellationToken, Task<int>>)(async (CancellationToken cs) => await GetResultWithMultipleExceptions(i, cs)));
+            var myEnum = new MyAsyncEnumerable<int>(funcs);
+            var initialList = Enumerable.Range(1, 50).ToList();
```

Comment:
Кажется, тут лишний ToList:

```suggestion
            var initialList = Enumerable.Range(1, 50);
```